src/tools/lib/python/particleCalorimeter.py

```diff
@@ -0,0 +1,245 @@
+            + self.stepStr + "_0_0_0.h5"
+
+        if not os.path.isfile(self.calo_file):
+            print(
+                "Error: The output HDF5 file " +
+                self.calo_file +
+                " does not exist."
+            )
+            return
+
+        f = h5py.File(self.calo_file, 'r')
+
+        self.step = step
+        # calorimeter data
+        self.calo_data = np.array(f['/data/' +
+                                    self.stepStr + "/" +
+                                    self.attribute_name])
+        # simulation unit
+        self.sim_unit = f['/data/' +
+                          self.stepStr + "/" +
+                          self.attribute_name].attrs['unitSI']
+
+        # cleanup
+        f.close()
+
+    def energy_histogram(self):
+        """
+        Creates an energy histogram that can be compared to the output of the
+        `energyHistogram` plugin output. The data is therefore summed
+        over the yaw and pitch angles.
+        """
+        self.hist_data = np.sum(self.calo_data, axis=(1, 2))
+
+    def plot_histogram(self,
+                       with_outliers=False,
+                       norm=False):
+        """
+        Parameters
+        ----------
+        with_outliers : boolean
+            plots also underflow and overflow bins if ``True``
+        norm : boolean
+            normalizes the histogram to the total number of counts
+            in all bins if ``True``
+        """
+        # unit conversion: Joules to eV
+        J2EV = 1.602e-19
+        # unit conversion: simUnit to EV
+        simUnitEV = self.sim_unit / J2EV
+
+        figHist = plt.figure(0)
+        axHist = figHist.add_subplot(111)
+
+        # absolute histogram data [photon count]
+        abs_hist_data = self.hist_data * \
+            simUnitEV / 1e3 / self.energy_range
+        energy_range = self.energy_range
+
+        # remove outliers from data to plot
+        if with_outliers is False:
+            abs_hist_data = abs_hist_data[1:-1]
+            energy_range = energy_range[1:-1]
+
+        if norm is False:
+            axHist.plot(energy_range,
+                        abs_hist_data,
+                        label="{}".format(int(self.step)))
+        else:
+            normed_hist_data = abs_hist_data / np.sum(abs_hist_data)
+            axHist.plot(energy_range,
+                        normed_hist_data,
+                        label="{}".format(int(self.step)))
+
+        axHist.set_xscale("log")
+        axHist.set_yscale("log")
+        axHist.set_xlabel("energy [keV]")
+        axHist.set_ylabel("count [arb. u.]")
+        axHist.set_title("`{}` energy histogram".format(self.species_name))
+        axHist.set_xlim(.5 * self.min_energy, 2 * self.max_energy)
+        axHist.legend(loc="best", title="counts after # steps")
+
+        plt.draw()
+        plt.show()
+
+    def plot_calorimeter(self):
+        """
+        Plots an imshow of the particle calorimeter data.
```

Comment:
Thx but this is not relevant here since the calorimeter data will be summed over all energies and on the axes you will find the opening angles. :)